File: sections/14_core_patterns_mixed_contest/problems/j_interval_point_cover/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int n;
    cin >> n;
    vector<pair<long long, long long>> intervals(n);
    for (auto &[left, right] : intervals) cin >> left >> right;
    sort(intervals.begin(), intervals.end(),
         [](auto a, auto b) { return tie(a.second, a.first) < tie(b.second, b.first); });
    long long last = LLONG_MIN;
    int answer = 0;
    for (auto [left, right] : intervals) {
        if (last < left) {
            last = right;
            ++answer;
        }
    }
    cout << answer << '\n';
}

File: sections/14_core_patterns_mixed_contest/problems/j_interval_point_cover/solution.py
import sys


def main() -> None:
    data = list(map(int, sys.stdin.buffer.read().split()))
    intervals = list(zip(data[1::2], data[2::2]))
    intervals.sort(key=lambda interval: (interval[1], interval[0]))
    last = None
    answer = 0
    for left, right in intervals:
        if last is None or last < left:
            last = right
            answer += 1
    print(answer)


if __name__ == "__main__":
    main()
